### PR TITLE
Fix for https://github.com/SteveSandersonMS/WebWindow/issues/88

### PR DESCRIPTION
**Pull request by jspuij**

### CODE REVIEW

File: src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            this.fatalError = error.toString();
+            console.error(`There was an error applying batch ${receivedBatchId}.`);
+
+            // If there's a rendering exception, notify server *and* throw on client
+            DotNet.invokeMethodAsync('Microsoft.MobileBlazorBindings.WebView', 'OnRenderCompleted', receivedBatchId, error.toString());
+            throw error;
+        }
+    }
+
+    public getLastBatchid(): number {
```
Comment (Eilon) [inline src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts:55]: This method isn't called? Did you mean to use it elsewhere in this type where the same `id - 1` logic is used?

File: src/Microsoft.MobileBlazorBindings.WebView/BlazorHybridRenderer.cs [hunk 2]
```diff
@@ -51,6 +58,65 @@ public BlazorHybridRenderer(IPC ipc, IServiceProvider serviceProvider, ILoggerFa
...
 
+        /// <summary>
+        /// Signals that a render batch has completed.
+        /// </summary>
+        /// <param name="incomingBatchId">The render batch id.</param>
+        /// <param name="errorMessageOrNull">The error message or null.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task OnRenderCompletedAsync(long incomingBatchId, string errorMessageOrNull)
+        {
+            if (this._disposing)
```
Comment (Eilon) [inline src/Microsoft.MobileBlazorBindings.WebView/BlazorHybridRenderer.cs:69]: Not a big deal right now, but we avoid `this.`. I thought the editorConfig in the file might catch this but I don't recall. Feel free to update this, or I will eventually do another cleanup pass in the future anyway (and probably break something while I'm at it).

File: src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿import '@dotnet/jsinterop/dist/Microsoft.JSInterop';
+import { renderBatch } from '@browserjs/Rendering/Renderer';
+import { OutOfProcessRenderBatch } from '@browserjs/Rendering/RenderBatch/OutOfProcessRenderBatch';
+
+export class RenderQueue {
+    private static instance: RenderQueue;
+
+    private nextBatchId = 2;
```
Comment (Eilon) [inline src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts:8]: Why does this start at 2? Is that a magic number?

File: src/Microsoft.MobileBlazorBindings.WebView.JS/src/Boot.Desktop.ts [hunk 4]
```diff
@@ -33,9 +33,9 @@ function boot() {
     DotNet.jsCallDispatcher.endInvokeDotNetFromJS(callId, success, resultOrError);
   });
 
-  ipc.on('JS.RenderBatch', (rendererId, batchBase64) => {
+    ipc.on('JS.RenderBatch', (batchId, batchBase64) => {
```
Comment (Eilon) [inline src/Microsoft.MobileBlazorBindings.WebView.JS/src/Boot.Desktop.ts:36]: Extra indent by accident?

File: src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts (hunk 3 above)
Comment (jspuij) [reply src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts:8]: Yeah, the first one that comes from C# is number one...

Comment (Eilon) [reply src/Microsoft.MobileBlazorBindings.WebView.JS/src/RenderQueue.ts:8]: I think this deserves a comment. The only numbers that don't require a comment are -MAX, -1, 0, 1, and MAX. This isn't one of those 😄

**Comment by Eilon** [comment]
Thank you again!

**Pull request merged**